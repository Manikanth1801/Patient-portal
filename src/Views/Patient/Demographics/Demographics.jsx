import React, { Component } from 'react';
import { AvForm, AvField} from 'availity-reactstrap-validation';
import axios from 'axios';


class Demographics extends Component {
    constructor(props) {
        super(props);

        this.state = {
            medicalHistory: '',
            familyMedicalHistory: '',
            surgeries: '',
            insuranceProvider: ''
        }
    }
    handleChange = (e) => {
        // console.log('e',e.target.value)
        this.setState({
            [e.target.name]: e.target.value
        })
    };

    handleSubmit = (e,error) => {
        if (error == '') {
            let demoData = {

                medicalHistory: this.state.medicalHistory,
                familyMedicalHistory: this.state.familyMedicalHistory,
                surgeries: this.state.surgeries,
                insuranceProvider: this.state.insuranceProvider

            }
            console.log('demoData', demoData);
            axios.post("http://localhost:8000/patientDemographics", demoData)
                .then(res => {
                    console.log('res', res)
                    if (res.data) {
                        alert("Demographics update is Success.")
                    }
                })
                .catch(err => {
                    if (err) {
                        alert("something went wrong.Please try after some time")
                        console.log(err);
                    }
                })
        }

    }

    render() {
        return (
            <div >

                <AvForm onSubmit={this.handleSubmit}>
                    <div className="row">
                        <div className="col-6">
                            <label className="h6">Medical History</label>
                        </div>
                        <div className="form-group col-8">
                            <AvField type='textarea' name="medicalHistory" value={this.state ? this.state.medicalHistory : ''} onChange={this.handleChange} id="medicalHistory" validate={{
                                required: { value: true, errorMessage: "Medical History is required" },

                            }}
                            />
                        </div>
                    </div>
                    <div className="row">
                        <div className="col-6">
                            <label className="h6">Family Medical History</label>
                        </div>
                        <div className="form-group col-8">
                            <AvField type='textarea' name="familyMedicalHistory" value={this.state ? this.state.familyMedicalHistory : ''} onChange={this.handleChange} id="familyMedicalHistory" validate={{
                                required: { value: true, errorMessage: "Family Medical History is required" },

                            }} />
                        </div>
                    </div>
                    <div className="row">
                        <div className="col-6">
                            <label className="h6">Surgeries</label>
                        </div>
                        <div className="form-group col-8">
                            <AvField type='textarea' name="surgeries" value={this.state ? this.state.surgeries : ''} onChange={this.handleChange} id="surgeries" validate={{
                                required: { value: true, errorMessage: "Surgeries is required" },

                            }} />
                        </div>
                    </div>
                    <div className="row">
                        <div className="col-6">
                            <label className="h6">Insurance Provider</label>
                        </div>
                        <div className="form-group col-8">
                            <AvField type='textarea' name="insuranceProvider" value={this.state ? this.state.insuranceProvider : ''} onChange={this.handleChange} id="insuranceProvider" validate={{
                                required: { value: true, errorMessage: "Insurance Provider is required" },

                            }} />
                        </div>
                    </div>
                    <br />
                    <div className="row">
                        <div className="col-6" id='submit'>
                            <button className="btn btn-outline-primary">Submit</button>
                        </div>
                    </div>
                </AvForm>
            </div>





        )
    }
}

export default Demographics