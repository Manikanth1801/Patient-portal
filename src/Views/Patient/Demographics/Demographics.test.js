import { shallow } from "enzyme";
import { describe, expect, test, it } from '@jest/globals';
import Demographics from "./Demographics";

describe('Test Cases for Demographics component',()=>{
    const onClickHandler = jest.fn();
    const demographics = shallow(<Demographics onClick={onClickHandler} />);

    it('should render properly', () => {
        expect(demographics.getElements()).toMatchSnapshot()
    })

    it('should have one AvField[type="textarea"] element', () => {
        const input = demographics.find('AvField[type="textarea"]').length;
        expect(input).toEqual(4)

    })
    it('validates model on button click', () => {
        it('should have button for update the form', () => {
            const buttonElement = demographics.find("button");
            buttonElement.simulate("click");
            expect(onClickHandler).toBeCalledTimes(1);
        })
      });



})