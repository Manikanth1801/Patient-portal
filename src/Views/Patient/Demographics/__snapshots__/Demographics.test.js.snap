// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Cases for Demographics component should render properly 1`] = `
Array [
  <div>
    <AvForm
      method="get"
      model={Object {}}
      onInvalidSubmit={[Function]}
      onKeyDown={[Function]}
      onSubmit={[Function]}
      onValidSubmit={[Function]}
      tag={[Function]}
      validationEvent={
        Array [
          "onChange",
          "onInput",
        ]
      }
    >
      <div
        className="row"
      >
        <div
          className="col-6"
        >
          <label
            className="h6"
          >
            Medical History
          </label>
        </div>
        <div
          className="form-group col-8"
        >
          <AvField
            id="medicalHistory"
            name="medicalHistory"
            onChange={[Function]}
            type="textarea"
            validate={
              Object {
                "required": Object {
                  "errorMessage": "Medical History is required",
                  "value": true,
                },
              }
            }
            value=""
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-6"
        >
          <label
            className="h6"
          >
            Family Medical History
          </label>
        </div>
        <div
          className="form-group col-8"
        >
          <AvField
            id="familyMedicalHistory"
            name="familyMedicalHistory"
            onChange={[Function]}
            type="textarea"
            validate={
              Object {
                "required": Object {
                  "errorMessage": "Family Medical History is required",
                  "value": true,
                },
              }
            }
            value=""
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-6"
        >
          <label
            className="h6"
          >
            Surgeries
          </label>
        </div>
        <div
          className="form-group col-8"
        >
          <AvField
            id="surgeries"
            name="surgeries"
            onChange={[Function]}
            type="textarea"
            validate={
              Object {
                "required": Object {
                  "errorMessage": "Surgeries is required",
                  "value": true,
                },
              }
            }
            value=""
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-6"
        >
          <label
            className="h6"
          >
            Insurance Provider
          </label>
        </div>
        <div
          className="form-group col-8"
        >
          <AvField
            id="insuranceProvider"
            name="insuranceProvider"
            onChange={[Function]}
            type="textarea"
            validate={
              Object {
                "required": Object {
                  "errorMessage": "Insurance Provider is required",
                  "value": true,
                },
              }
            }
            value=""
          />
        </div>
      </div>
      <br />
      <div
        className="row"
      >
        <div
          className="col-6"
          id="submit"
        >
          <button
            className="btn btn-outline-primary"
          >
            Submit
          </button>
        </div>
      </div>
    </AvForm>
  </div>,
]
`;
